[New line] STM32MP1xx (just STM32CubeMP1 package)

kbumsik opened the pull request:
**Summary**

<!-- Summary of the PR -->

This PR adds STM32CubeMP1 package. In detail this adds the following:

* [ ] STM32CubeMP1 CMSIS
* [ ] STM32CubeMP1 HAL
* [ ] OpenAMP middleware from STM32CubeMP1. This is used for serial-like communications between the host MPU (Cortex-A7) and the Arduino MCU (Cortex-M4).
* [ ] Some simple and tedious fixes for Arduino Core 

I'm currently working on STM32MP157 Discovery boards port. While the work is pretty much done and I will make a new PR for it (hopefully in this week), for now I would like to make a separate PR that just adds a STM32CubeMP1 package so that it will reduce excessive noises in the next PR.

I pulled the STM32CubeMP1 v1.0.1 from its [Github release](https://github.com/STMicroelectronics/STM32CubeMP1/releases/tag/1.0.1) directly.

fpistm commented:
Hi @kbumsik 
Thanks for this PR, I will review it soon.

fpistm commented:
For Astyle issue, you should add `system/Middlewares/OpenAMP/`
in https://github.com/stm32duino/Arduino_Core_STM32/blob/master/CI/astyle/.astyleignore

kbumsik commented:
Thanks for mentioning about Astyle.

BTW I need to discuss one thing. As STM32MP1 is indeed a special board among other STM32 MCUs, it is necessary to include a new README document on how to use STM32MP1 Discovery board with Arduino IDE. For example, instructions about a special upload tool (STM32Prog is not supported) and some warnings regarding collision of Arduino binary and Linux device tree.

Where is the best place to put the READMEs? Maybe in `Arduino_Core_STM32/variants/__STM32MP1_Boards__`?

fpistm commented:
> Where is the best place to put the READMEs? Maybe in `Arduino_Core_STM32/variants/__STM32MP1_Boards__`?

Good question, I guess, as a first step, this could be add in the variant.

fpistm commented:
Hi  @kbumsik 

> I'm currently working on STM32MP157 Discovery boards port. While the work is pretty much done and I will make a new PR for it soon (hopefully in this week)

Do you succeed?

kbumsik commented:
@fpistm Pretty much, except I2C is not working properly for unknown reason and that is the only reason I haven't made a PR for weeks :(
I think it's the time I should just open a PR as a WIP for now. I will open it after the weekend. Are you planning to review the current PR after that?

fpistm commented:
OK thanks for the quick feedback.
I will try to get a board then I will be able to review and test.

kbumsik commented:
Great, I got another person who can test the actual hardware :D I will make a PR then.

fpistm commented:
Hi @kbumsik 
I will made a new PR with  the latest CubeFw and updated path as the HAL/LL source files are now in a built-in library.

aedancullen commented:
If you need another person to test on hardware, I also have a STM32MP157C-DK2 board and will be working on getting this up and running.

fpistm commented:
Hi @aedancullen 
thanks for your proposition, any help are welcome.
I will update the MP1 Cube package because there is a new version and the way it is integrated in the core has changed since this PR was done.
Anyway, @kbumsik variant is always correct, so I'm waiting his PR for the variant.

kbumsik commented:
@fpistm @aedancullen I was quite busy for a moment but now I can make a PR: #717 :)
I updated the Cube library to 1.1.0.
The weird thing is, that the some issues with peripherals like I2C I mentioned suddenly disappeared after I updated the Cube library and rebasing it to the latest master. Now all peripherals looks fine...😮 I don't know what happened but it is a good thing anyway.

#717 also has the same commits as this PR. I will clean up #717 after this PR is merged.

fpistm commented:
I think I found the issue they made some update after GitHub release 1.1.0:
https://github.com/STMicroelectronics/STM32CubeMP1/commit/0e09d0c295134543ba5170f7941f1b8684816113
